railties/lib/rails/commands/dbconsole/dbconsole_command.rb

```diff
@@ -99,6 +99,10 @@ def environment
       Rails.respond_to?(:env) ? Rails.env : Rails::Command.environment
     end
 
+    def connection
+      @options[:connection] || "primary"
```

Comment:
Actually I added it back because `Rails::DBConsole` is public API so people may be using it directly, without Thor. 😊